src/cls/rgw/cls_rgw.cc

```diff
@@ -3172,10 +3172,10 @@ static int gc_iterate_entries(cls_method_context_t hctx, const string& marker, b
     *truncated = false;
 
   string start_key;
-  if (key_iter.empty()) {
+  if (marker.empty()) {
     prepend_index_prefix(marker, GC_OBJ_TIME_INDEX, &start_key);
   } else {
-    start_key = key_iter;
+    start_key = marker;
```

Comment:
okay, so the first call with an empty marker will use `prepend_index_prefix()` to set `start_key = "1_"`, then any markers returned in `key_iter` will already have that prefix